src/software/new_geom/bezier_curve2d_test.cpp

```diff
@@ -0,0 +1,144 @@
+#include "software/new_geom/bezier_curve2d.h"
+
+#include <gtest/gtest.h>
+
+#include "software/test_util/test_util.h"
+
+using namespace Test;
+
+TEST(BezierCurve2dTest, contstruct_with_less_then_two_control_points)
+{
+    std::vector<Point> points = {Point(0, 0)};
+
+    EXPECT_THROW(BezierCurve2d curve(points), std::invalid_argument);
+}
+
+TEST(BezierCurve2dTest, get_value_two_points)
+{
+    // Linear bezier curve
+    Point p0(-1, 2);
+    Point p1(5, -7);
+    const std::vector<Point> points = {p0, p1};
+
+    const BezierCurve2d curve(points);
+
+    const Polynomial2d expected_polynomial2d({p0, p1});
+
+    // We approximate an equality check by sampling at a bunch of points and comparing
+    // too the expected polynomial
+    for (int i = 0; i <= 1000; i++)
+    {
+        const double t = static_cast<double>(i) / 1000.0;
+        EXPECT_TRUE(TestUtil::equalWithinTolerance(expected_polynomial2d.getValueAt(t),
+                                                   curve.getValueAt(t), 1e-9));
+    }
+}
+
+TEST(BezierCurve2dTest, get_value_below_zero)
+{
+    // Linear bezier curve
+    Point p0(-1, 2);
+    Point p1(5, -7);
+    const std::vector<Point> points = {p0, p1};
+
+    const BezierCurve2d curve(points);
+
+    // Check that getting a value below zero just returns the first point
+    EXPECT_EQ(p0, curve.getValueAt(-0.1));
+}
+
+TEST(BezierCurve2dTest, get_value_above_one)
+{
+    // Linear bezier curve
+    Point p0(-1, 2);
+    Point p1(5, -7);
+    const std::vector<Point> points = {p0, p1};
+
+    const BezierCurve2d curve(points);
+
+    // Check that getting a value just above one just returns the last point
+    EXPECT_EQ(p1, curve.getValueAt(1.1));
+}
+
+TEST(BezierCurveTest, get_value_four_points)
+{
+    // Cubic bezier curve
+    Vector v0(-1, 2);
```

Comment:
nit: why construct vectors instead of points here?

Reply:
Whoops, this was carried over from an old version of the tests. Changed all to points.